tikv-client/src/main/java/com/pingcap/tikv/types/DecimalType.java

```diff
@@ -48,6 +48,7 @@ private DecimalType(int tp) {
    */
   @Override
   public Object decodeNotNull(int flag, CodecDataInput cdi) {
+    super.decodeNotNull(flag, cdi);
```

Comment:
I don't get what this line really does.

Reply:
I don't think this is right either. @zhexuany Would you please explain a little bit?

Reply:
@ilovesoup Yes. It is of course not right. Original way was my intention to let `DecimalType`  has ability to decode `RealType`. Please refer to my discussion with bridstorm. We both agreed we need let `Expr` carrying `ColumnInfo`. In such way, we can use either a if-else statement or switch to complete sum logic. For you information, sum logics has been changed a little bit against tidb's  master. Originally, everything was promoted to decimal to handle. Right now, only int and decimal will keep same way.  float will return float.

Reply:
I will raise another pr to add config about TiKV version, so we could decide on which logic to choose. What do you think? @zhexuany

Reply:
It is better to do in same PR.  Till type refectory merged into master, I will finish this PR.